Makefile

```diff
@@ -26,5 +29,8 @@ clean: $(CLEAN_DIRS)
 $(CLEAN_DIRS):
 	$(MAKE) -C $(@:clean-%=%) clean
 
+unit: $(UNITTEST_DIR)/Makefile
```

Comment:
I don't think it should be dependent on the unit Makefile, because a .o could be added without touching the makefile.  (and the same comment for a similar dependency in test/unit/Makefile).  I'm not sure if there's an easy way to track the .o changes/additions?  Maybe you would know, your Makefile-fu seems pretty strong :-)

I think it's probably fine for now to just remove any dependencies and run the unit tests every time.

Reply:
It also might make sense to add a dependency on create-diff-object here.

Reply:
(but only if we also figured out how to make a dependency on the .o files, I think)

Reply:
That dependency does not affect when it runs because `unit` is not an automatic target(and I would argue against making it one because of git-lfs dependency), so it will always run when invoked by `make unit`. $(UNITTEST_DIR)/Makefile is just a dependency that make doesn't know how to create so it will fail early if it is not there.

It is, however a good idea to make `unit` a PHONY target and add $(BUILD_DIRS) dependency so it will automatically build everything if it is not built yet.

It would also be a good idea to add create-diff-object dependency to test/unit/Makefile.include to the rules where it is used, that way it will retest everything on next `make unit` invocation if create-diff-object was rebuilt.

Reply:
That all sounds good.  But if I run 'make unit' twice, the second run doesn't do anything.  Is that what we want?  I presume that means that if we add a new .o file, it won't detect it and rerun the tests, right?

Reply:
(And if so, maybe that's ok, we'll just have to remember to `make clean` before running the tests again, after adding a .o file)

Reply:
> That all sounds good. But if I run 'make unit' twice, the second run doesn't do anything. Is that what we want?

If you touch any of the .ORIG.o/.PATCHED.o it would retest the touched testcase and it would detect new ones as well. When you do `make unit` it unconditionally ends up in test/unit/$(uname -m)/Makefile that has all of the automated targets and the only dependency logic that matters is there.